Feature/network health monitoring

theodesu opened the pull request:
Add network monitoring features of Mira  v. 2.10. Provisioned dashboard in grafana now looks like:
![image](https://github.com/user-attachments/assets/9e975a5e-0c9d-4696-8636-e37dc9beba32)
![image](https://github.com/user-attachments/assets/bed8a159-f6af-4f17-84ef-fc0288a58d8c)

To achieve this I've:
* Implemented new mira C-API in the ruuvi FW, with configuration over NFC.
* Updated udp_to_influx.py to use Influx v2
* Updated grafana data source plugin to use influx v2
* Created new bucket for ruuvi sensor data in existing influx v2 instance for monitoring data from Mira deployment files
* Included installation of mira-gateway and mira-monitoring services in shell script, for (almost) a one-line install.
* Updated dashboard json to new format, using flux query language instead of JQL (and solved the bug of data sources not being deployable through docker compose)

Future improvements:
Monitoring C-API is hardcoded to send all sensor values, but they aren't used in the current dashboard. To optimise, one could choose which ones are relevant and only send them. Or make the bitfield configurable over NFC.

Commit:
add network monitoring feature to firmware

Commit:
Add support for mira-monitoring in grafana dashboard

Merges ruuvi-tag data and monitoring data into the same influx instance.
Bumps udp_to_influx to use influx2.
Updates dashboard to display mira-monitoring data.
Makes dashboard deployable with datasources without manual intervention.
One docker compose for all containers, including monitoring stuff.
One install script for everything.

bofh69 commented on RPi/install_dashboard.sh:
```diff
@@ -0,0 +1,33 @@
+        exit 1
+fi
+
+arch=arm64
```
Maybe use $(uname -m) insted?

theodesu replied:
This returns "aarch64" on 64bit arm. I could set arch explicitly on a condition of this return, which at least makes it automatic.

bofh69 commented on RPi/mira-gateway.env:
```diff
@@ -0,0 +1,2 @@
+MIRA_GW_PAN_ID=13243546
+MIRA_GW_KEY=11121314212223243132333441424344
```
Should the install-script create these instead?

mira_gen_panid.py could be used to generate the pan (might need to be transformed with `printf "%08x" $(mira_gen_panid.py)`).

Hexdump can generate the key:
hexdump -n 16 -e '"%02x"' /dev/urandom

theodesu replied:
I added generation if libmira is vendored, otherwise it uses these credentials. I didn't want to create a hard dependency on libmira, and I also didn't want to copy mira_gen_panid.py to this repo.

Rob1nJ commented on RPi/services/udp-to-influx/udp_to_influx2.py:
```diff
@@ -233,6 +236,15 @@ def parse_ipv6_address(string):
 
 
 def main():
+    # parser = argparse.ArgumentParser(description="UDP to influx2)
+    # token = ""
+    # parser.add_argument(
+    #     "-t",
+    #     "--token",
+    #     dest=token,
+    #     help="Influxdb2 auth token",
+    #     required=True)
+    
```
Since this is not used, can it be removed?

theodesu replied:
I forgot about that! Had an idea but solved it in another way.

Rob1nJ commented on RPi/services/udp-to-influx/udp_to_influx2.py:
```diff
@@ -233,6 +236,15 @@ def parse_ipv6_address(string):
+    #     help="Influxdb2 auth token",
+    #     required=True)
+    
     root_addr = os.environ.get("ROOT_ADDR", None)
```
The README.md do not mention this environment variable anymore. If the variable is not needed then I think this  can be removed?

Rob1nJ commented on README.md:
```diff
@@ -13,9 +13,9 @@ RuuviTags form a Mira mesh network, transmitting sensor data to a border gateway
+1. libmira, version 2.10.0 or higher
 2. MiraUSB
-3. Raspberry Pi 3 or 4 with a SD card and a power supply
+3. Raspberry Pi 3, 4 or 5 with a SD card and a power supply
```
Influxdb2 is only available for arm64.

Commit:
Fix review comments

Only support 64 bit operating systems
Remove calculate_hops functionality
Generate credentials
